org.sonarlint.eclipse.cdt/src/org/sonarlint/eclipse/cdt/internal/SonarCdtPlugin.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * SonarLint for Eclipse
+ * Copyright (C) 2015-2016 SonarSource SA
+ * sonarlint@sonarsource.com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonarlint.eclipse.cdt.internal;
+
+import org.eclipse.core.resources.IProject;
+import org.eclipse.cdt.core.model.CoreModel;
+import org.sonarlint.eclipse.core.AbstractPlugin;
+
+public class SonarCdtPlugin extends AbstractPlugin {
+
+  public static final String PLUGIN_ID = "org.sonarlint.eclipse.cdt"; //$NON-NLS-1$
+
+  private static SonarCdtPlugin plugin;
+
+  public SonarCdtPlugin() {
+    plugin = this;
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Synchronize this lazy initialization of 'plugin' [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS2444)